base/abstractarray.jl

```diff
@@ -546,7 +546,7 @@ julia> stride(A,3)
 function stride(A::AbstractArray, k::Integer)
     st = strides(A)
     k ≤ ndims(A) && return st[k]
-    return sum(st .* size(A))
+    return ndims(A) == 0 ? 1 : sum(st .* size(A))
```

Comment:
This seems like a very low-level function to be calling broadcast and reduce. Should probably be a loop.

Reply:
I suggest restoring this function exactly:
```julia
function _stride(a, i)
    if i > ndims(a)
        return length(a)
    end
    s = 1
    for n = 1:(i-1)
        s *= size(a, n)
    end
    return s
end
```

https://github.com/JuliaLang/julia/blob/dd0c14ba1d0add2ce89524a26684a1194a83312c/base/reinterpretarray.jl#L162-L171

Reply:
The orginal version don't thrown a error for `stride(a, -1)`.
I think it's meaningful the keep the consistency of error message. The easist way is follow the implement in general fallback.
(And they should have no performance difference)